src/hooks/useContentFilters.ts

```diff
@@ -0,0 +1,28 @@
+import { useDispatch } from 'react-redux';
+import { Dispatch } from '../utils/types';
+import { requestFilter,
+  requestContent,
+  requestActiveButton,
+  requestQuantityNews,
+} from '../redux/actions';
+
+const useContentFilters = () => {
+  const dispatch = useDispatch<Dispatch>();
+
+  const handleContent = (e: React.MouseEvent<HTMLButtonElement>) => {
+    const { value } = e.target as HTMLButtonElement;
+    dispatch(requestActiveButton(value));
+    dispatch(requestQuantityNews(12));
+    if (value === 'favoritos') {
+      dispatch(requestFilter(value));
+      dispatch(requestContent(false));
+    } else {
+      dispatch(requestFilter(value));
+      dispatch(requestContent(true));
+    }
+  };
+
+  return { handleContent };
+};
```

Comment:
É interessante modelar as ações para não serem apenas "setters" de estado e sim eventos do usuário que resultam em mudanças no estado, como recomendado [nesse tópico do style guide do redux](https://redux.js.org/style-guide/#model-actions-as-events-not-setters).

> Writing "setters" often results in too many individual action types, too many dispatches, and an action log that is less meaningful.

O debug fica mais difícil dessa forma. 

A alternativa é ter apenas uma ação, ou até duas, já que tem uma lógica ali de favoritos. Uma sugestão: 

- `changeFilterToContent => { ...state, activeButton: value, quantityNews: 12, filter: value, content: true  }`
- `changeFilterToFavorites =>  { ...state, activeButton: value, quantityNews: 12, filter: value, content: false  }`

Ou simplesmente:

```ts
// no component
dispatch(changeFilter(value))
// no reducer
switch(state, action) {
  case CHANGE_FILTER:
     const { value } = action.payload
     const content = value === 'favoritos'
     return { ...state, activeButton: value, quantityNews: 12, filter: value, content  }
}
```

A última sendo bem interessante por trazer a lógica para dentro do reducer, como sugerido em outro [tópico da style guide](https://redux.js.org/style-guide/#put-as-much-logic-as-possible-in-reducers).

> This helps ensure that more of the actual app logic is easily testable, enables more effective use of time-travel debugging, and helps avoid common mistakes that can lead to mutations and bugs.

Reply:
Obrigado pela recomendação, realmente faz muito mais sentido. Movi a lógica para o reducer e diminuí a quantidade de actions.